test/unit/test_diagram.py

```diff
@@ -18,7 +18,15 @@
 # along with Autosubmit.  If not, see <http://www.gnu.org/licenses/>.
 
 """ Test file for autosubmit/monitor/diagram.py """
+
+import tempfile
 import datetime
+from pathlib import Path
+
+from mock.mock import patch
```

Comment:
`tepmfile` and `patch` can be replaced by the built-in fixtures that come with pytest: https://docs.pytest.org/en/6.2.x/fixture.html#pytest-fixtures-explicit-modular-scalable

For `tempfile` there's `tmp_path` which returns a `Path` already for you (I think you may not need to import Path` here, maybe.

And for `patch`, the `pytest_mock` lib we use has `mocker`, https://pytest-mock.readthedocs.io/en/latest/usage.html

Both work the same, but with pytest fixture the code is a bit shorter (one of the main selling points of pytest over unittest was to write less to have the same -- no classes, just functions, no need to import a lot of things, just use fixtures, etc...)

Reply:
I've recently written a few tests for `stats` and `monitor` in #2322 , and there you can see the `mocker.patch` fixture being used.